solidity/contracts/zeto_anon_enc_nullifier.sol

```diff
@@ -35,21 +36,27 @@ import "hardhat/console.sol";
 ///        - the nullifiers represent input commitments that are included in a Sparse Merkle Tree represented by the root hash
 contract Zeto_AnonEncNullifier is
     ZetoNullifier,
-    ZetoFungibleWithdrawWithNullifiers
+    ZetoFungibleWithdrawWithNullifiers,
+    UUPSUpgradeable
 {
     Groth16Verifier_AnonEncNullifier verifier;
 
-    constructor(
+    function initialize(
+        address authority,
```

Comment:
Can we name this `initialOwner`? Suppose we introduce a new role that can diverge from the initial contract owner (e.g. when a contract is upgraded, the authority remains unchanged). In that case, we probably should document `authority` is always the initial contract owner rather than hide the "owner" concepts.